### PR TITLE
Ported trio_matrix test to 0.2

### LINKED ISSUE #2814: Convert trio matrix test to api2
Deleting in col key branch.

```

    def test_trio_matrix(self):
        """
        This test depends on certain properties of the trio matrix VCF
        and pedigree structure.

        This test is NOT a valid test if the pedigree includes quads:
        the trio_matrix method will duplicate the parents appropriately,
        but the genotypes_table and samples_table orthogonal paths would
        require another duplication/explode that we haven't written.
        """
        ped = Pedigree.read(test_file('triomatrix.fam'))
        famkt = KeyTable.import_fam(test_file('triomatrix.fam'))

        vds = hc.import_vcf(test_file('triomatrix.vcf'))\
                .annotate_samples_table(famkt, root='sa.fam')

        dads = famkt.filter('isDefined(pat_id)')\
                    .annotate('is_dad = true')\
                    .select(['pat_id', 'is_dad'])\
                    .key_by('pat_id')

        moms = famkt.filter('isDefined(mat_id)') \
            .annotate('is_mom = true') \
            .select(['mat_id', 'is_mom']) \
            .key_by('mat_id')

        # test genotypes
        gkt = (vds.genotypes_table()
                  .key_by('s')
                  .join(dads, how='left')
                  .join(moms, how='left')
                  .annotate('is_dad = isDefined(is_dad), is_mom = isDefined(is_mom)')
                  .aggregate_by_key('v = v, fam = fam.fam_id',
                                    'data = GT.map(_ => {role: if (is_dad) 1 else if (is_mom) 2 else 0, g: {GT: GT, AD: AD, GQ: GQ, DP: DP, PL: PL}}).collect()')
                  .filter('data.length() == 3')
                  .explode('data')
                  .select(['v', 'fam', 'data']))

        tkt = (vds.trio_matrix(ped, complete_trios=True)
                  .genotypes_table()
                  .annotate('fam = proband.fields.fam.fam_id, data = [{role: 0, g: proband}, {role: 1, g: father}, {role: 2, g: mother}]')
                  .select(['v', 'fam', 'data'])
                  .explode('data')
                  .filter('isDefined(data.g)')
                  .key_by(['v', 'fam']))

        self.assertTrue(gkt.same(tkt))

        # test annotations
        g_sa = (vds.samples_table()
                   .join(dads, how='left')
                   .join(moms, how='left')
                   .annotate('is_dad = isDefined(is_dad), is_mom = isDefined(is_mom)')
                   .aggregate_by_key('fam = fam.fam_id',
                                     'data = map(sa => {role: if (is_dad) 1 else if (is_mom) 2 else 0, sa: sa}).collect()')
                   .filter('data.length() == 3')
                   .explode('data')
                   .select(['fam', 'data']))

        t_sa = (vds.trio_matrix(ped, complete_trios=True)
                .samples_table()
                .annotate('fam = sa.proband.fields.fam.fam_id, data = [{role: 0, sa: sa.proband.fields}, '
                          '{role: 1, sa: sa.father.fields}, '
                          '{role: 2, sa: sa.mother.fields}]')
                .select(['fam', 'data'])
                .explode('data')
                .filter('isDefined(data.sa)')
                .key_by(['fam']))

        self.assertTrue(g_sa.same(t_sa))
```

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #2814

### CODE REVIEW

File: python/hail/tests/utils.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 
 def startTestHailContext():
-    hail.init(master='local[2]', min_block_size=0, quiet=True)
+    hail.init(master='local[2]', min_block_size=0, quiet=False)
```
Comment (jbloom22) [inline python/hail/tests/utils.py]: intentional?

File: python/hail/tests/test_methods.py [hunk 2]
```diff
@@ -138,14 +138,74 @@ def test_linreg(self):
...
+            .join(moms, how='left'))
+        et = et.annotate(is_dad=hl.is_defined(et.is_dad),
+                         is_mom=hl.is_defined(et.is_mom))
+
+        et = (et
+            .group_by(et.locus, et.alleles, fam=et.fam.fam_id)
+            .aggregate(data=hl.agg.collect(hl.struct(
+            role=hl.case().when(et.is_dad, 1).when(et.is_mom, 2).default(0),
+            g=hl.struct(GT=et.GT, AD=et.AD, DP=et.DP, GQ=et.GQ, PL=et.PL)
+        ))))
```
Comment (jbloom22) [inline python/hail/tests/test_methods.py]: move parens up

File: python/hail/tests/test_methods.py [hunk 3]
```diff
@@ -138,14 +138,74 @@ def test_linreg(self):
         dataset.count_rows()
 
     def test_trio_matrix(self):
+        """
+        This test depends on certain properties of the trio matrix VCF
+        and pedigree structure.
+
+        This test is NOT a valid test if the pedigree includes quads:
```
Comment (jbloom22) [inline python/hail/tests/test_methods.py]: No need for new paragraph. Just say continue with "It is NOT a ..."

File: python/hail/tests/utils.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/tests/utils.py]: ah oops. I like quiet for python tests but some of the useful information in `_same` is silenced. I can try to add that to the exception it throws.

**Pull request merged**